draft-ietf-quic-spin-exp.md

```diff
@@ -238,6 +238,17 @@ and the observer and the client, respectively. It does this by measuring the
 delay between a spin edge observed in the upstream direction and that observed
 in the downstream direction, and vice versa.
 
+# Disabling the spin bit
+
+Implementation SHOULD allow users to disable the spin-bit either globally,
+or on a per-connection basis. Even when the spin-bit is not disabled by
+the user, implementations SHOULD disable the spin-bit on a randomly chosen
+fraction of connections. The random process SHOULD be designed so that
+on average the spin-bit is disabled for at least 1/8th of the connections.
+
+When the spin-bit is disabled, peers MUST set the spin bit value to zero,
```

Comment:
```suggestion
When the spin bit is disabled, endpoints MUST set the spin bit value to zero,
```

Any sense in permitting different forms of disabling?  I mean, there are other ways to erode the signal.  Fixed to zero, fixed to one, random, alternating values on every packet.  I wouldn't prohibit any (or all) of these techniques from being used.

Reply:
I would rather say that the endpoint can set the value to any value then.  Providing a concrete suggestion means that anyone who diverges from that suggestion will stick out.  Not providing a concrete suggestion means that you might end up with the choice of method being used to identify an implementation, but that's probably a lost cause.

Reply:
For greasing it might be (slightly) better to just use random values for this bit, otherwise middleboxes might learn that this bit is either 0 all the time or contains a valid spin bit.

Reply:
In order of complexity, I can see:

1) Always zero. Simple, people won't mess up.
2) Pick either 0 or 1 randomly at the beginning of the connection. Simple enough, somewhat better than always zero for preventing ossification.
3) Pick a random value each time. I don't like that too much. Implementation will mess up and generate a fingerprint.

I could specify flip a coin at the beginning of each connection if that's what you want.

Reply:
In New York we discussed the proposal to grease whatever bits are left in the first octet by just encrypting 0s. I assumed that peers opting out of the spin bit would use the same algorithm they use for greasing the other bits.

Reply:
@marten-seemann Yes, encryption is better than just randomization, especially if we specify it completely so there are no differences between implementations.

Reply:
You can't encrypt using the same technique because the value you choose will be input to the packet protection AEAD and so might alter the header encryption sampling.  If the intent is to appear random, a CSPRNG would be fine.  If the value can be predicted, then you risk undoing packet number encryption, promoting linkability, or something worse.